nemo_aligner/utils/trt_llm.py

```diff
@@ -0,0 +1,175 @@
+import secrets
+
+import tensorrt_llm
+import torch
+
+from nemo.export.tensorrt_llm import TensorRTLLM
+from nemo.export.trt_llm import tensorrt_llm_run
+from nemo.export.trt_llm.nemo_ckpt_loader.nemo_file import build_tokenizer
+from nemo.utils import logging
+from nemo_aligner.utils import parallel_state
+from nemo_aligner.utils.distributed import broadcast_2d_tensor_within_mp
+
+
+class GPTGenerateTRTLLM:
+    def __init__(
+        self,
+        model_cfg,
+        max_generation_length=1024,
+        max_input_len=1024,
+        max_input_tokens=4096,
+        generation_batch_size=4,
+        unload_engine_train=False,
+        trt_model_type="GPTForCausalLM",
+        end_strings=None,
+        reshard_model=False,
+        sample_temperature=None,
+        sample_top_k=None,
+        sample_top_p=None,
+        repetition_penalty=1.0,
+        use_greedy=False,
+        tokenizer=None,
+        trt_model_dir="/tmp/trt_llm_model",
+        seed=None,
```

Comment:
Definitely not a huge deal, but I'd rather have either sensible default values or make parameters mandatory than use `None` as default, when `None` is invalid (I haven't checked all settings, but for instance using `None` for `end_strings` or `tokenizer` is going to crash)

Reply:
added more sensible defaults 4526db8ff47047e3a9fac12e8e59a4f61a5aa73c

i didn't need to change anything other than this class since the caller fully specify all the arguments. I used the nemo sampling default parameters with the exception of use_greedy=False instead of True